### PR TITLE
Add workflow config to limit subject's height

### PR DESCRIPTION
**Pull request by goplayoutside3**
## Package
lib-classifier

## Describe your changes

For context, PFE's classifier displays subject images at their natural dimensions (at maximum), while in contrast, FEM is designed to stretch subject images to fit a volunteer's browser window.

This PR introduces the `limit_subject_height` property to the WorkflowConfiguration store and a corresponding Centered Layout component. There are three subject viewers where this could apply: SingleImageViewer, FlipbookViewer, and MultiFrameViewer. In each case, I passed the `limit_subject_height` property to the SVGPanZoom component to give it a max-height of either 90vh or the subject image's natural pixel height, whichever is smaller.

When a subject's height is limited, and its width does not fill a volunteer's browser, the CenteredLayout displays the classifier components center-justified. See more comments below about this decisions and its consequences.

## How to Review

The `limit_subject_height` property can be toggled with a checkbox in any project's fem-lab. For example, here's a test project:
- https://www.zooniverse.org/lab/14974/workflows/17600?femLab=true
- https://local.zooniverse.org:3000/projects/goplayoutside3/fem-test-project/classify/workflow/17600?env=production

The target project is Daily Minor Planet: https://local.zooniverse.org:3000/projects/fulsdavid/the-daily-minor-planet/classify/workflow/22354?env=production

I added a story to SingleImageViewer's storybook that handles `limit_subject_height`.
I also added a story for CenteredLayout.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [x] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [x] Can submit a classification
- [x] Can sign-in and sign-out
- [x] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
- [x] The PR creator has listed user actions to use when testing the new feature
- [x] Unit tests are included for the new feature
- [x] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

### CODE REVIEW

**Comment by goplayoutside3** [comment]
@lcjohnso I deployed this branch to get your initial thoughts on the UX. Here's an example project that has "display subjects with natural dimensions" enabled:
- FEM: https://fe-project-branch.preview.zooniverse.org/projects/goplayoutside3/fem-test-project/classify/workflow/17600
- Same project on PFE: https://www.zooniverse.org/projects/goplayoutside3/fem-test-project/classify
- Project's fem-lab: https://www.zooniverse.org/lab/14974/workflows/17600?femLab=true

Something I noticed immediately is that the image toolbar and meta-tools occupy a lot of space in FEM's design. I also noticed that many projects launched on PFE do not render an image toolbar at all, such as [Galaxy Zoo](https://www.zooniverse.org/projects/zookeeper/galaxy-zoo/classify). Thoughts on any UX adjustments to this feature?

**Comment by eatyourgreens** [comment]
I think [`aspect-ratio`](https://developer.mozilla.org/en-US/docs/Web/CSS/aspect-ratio) is well-supported in browsers now. Would that help here, if image aspect ratio is the problem?

**Comment by eatyourgreens** [comment]
>Something I noticed immediately is that the image toolbar and meta-tools occupy a lot of space in FEM's design. I also noticed that many projects launched on PFE do not render an image toolbar at all, such as [Galaxy Zoo](https://www.zooniverse.org/projects/zookeeper/galaxy-zoo/classify).

I can answer this. the toolbar is optional in PFE (toggled by the pan/zoom option for a workflow.) We made a decision to deprecate that in the new classifier. There might be an ADR for that decision, with details.

**Comment by eatyourgreens** [comment]
If you set the subject viewer grid column width to [`max-content`](https://developer.mozilla.org/en-US/docs/Web/CSS/max-content), instead of `100rem`, Galaxy Zoo looks something like this. The image won't grow any bigger, but it sits in the centre of the subject viewer. This is just playing around with the style rules in the browser's dev tools.

Screenshot of Galaxy Zoo: Cosmic Dawn with the subject viewer width set to max-content. The galaxy image is centered in the subject viewer, with gray margins on all sides.

**Comment by goplayoutside3** [comment]
@lcjohnso I modified this PR to calculated `zoom` based on the subject's natural dimensions. You can try it out on fe-project-branch.preview. I think the space around subjects is pretty annoying and caused by the subject viewer's dimension following the aspect ratio of the image. Examples:

Screenshot 2023-03-09 at 2 59 37 PM

Screenshot 2023-03-09 at 2 59 46 PM

1. If this `zoom` implementation does not seem like a good route, we could instead limit the height of the subject viewer to less than 100vh. This would essentially be replicating the original original behavior of PFE's subject viewer, and the fem-lab option copy would then be labeled "limit subject display height" rather than "use subject image original dimensions". The zoomable area would match the image's dimensions on first load. This is what PFE does, too.

2. A second option although maybe not ideal would be to put a button in the image toolbar with a feature “expand subject area” that toggles the viewer dimensions to expand to a user’s browser. Once expanded, the subject viewer would look as it does now on production, but on first load the subject displays as project teams want it to (natural image dimeneions, small zoomable area). That way transformation of the viewer dimensions happens only once. This implementation could have strange affects on drawing tools though.

Thoughts on our options?

**Comment by lcjohnso** [comment]
I agree -- the space above the image basically defeats the purpose of this feature: you still have to scroll down to see the full image, nevermind see the links / items below the image.

I like your option 1, but where the feature is labelled "Cap subject height: smaller of natural image height or viewport height" -- because I still want to handle case of GZ subject (small & square; 424x424 image) so that it isn't blown up beyond natural size.

Downside of this solution: centering -- PFE's behavior was to center the full SubjectArea+TaskArea horizontally, whereas the current solution keeps the default SubjectArea width and overall centering unchanged.  My preference would be to keep the TaskArea fixed (I like that it doesn't move even if subject image sizes change) but center the image horizontally within the default SubjectArea.

**Comment by goplayoutside3** [comment]
Sounds good! I'll work on getting that option into this PR, and will rope in Sean for input on design choices.

**Comment by goplayoutside3** [comment]
I updated this branch and the PR's description to reflect that it handles a subject's height when the "limit subject height" option is selected as `true` in the FEM lab.

@seanmiller26 tagging you here for a look at the layout: Here's an example of what a 600px-height subject looks like in both frontend apps.

fem

pfe

The "zoomable and panning" area is that of the subject image when it first loads - meaning that when a volunteers uses the image toolbar in the screenshot of FEM, the empty space to the left and right of the image remains static. Do you suggest any design changes to handle portrait-oriented or even square-oriented subjects? 

Adjusting the "zoomable" area is a heavy lift, but perhaps we could instead make some minor adjustments to the classifier layout when project teams want their subject images to always fit in the browser window, if you think space on either side of a subject looks a bit awkward.

**Comment by goplayoutside3** [comment]
Re: Sean's question about empty space on either side of the "zoomable and panning" area. Yes the subject image is cropped and those areas are not filled during zoom, rotate, pan etc.

We had a Slack call today and I'm going to update this PR when a classifier layout that is centered like PFE's classifier when `limit_subject_height` is enabled. We'll see how it looks on the branch deploy.

**Review by seanmiller26 — APPROVED** [review]
As per our discussion this centering alignment makes the most sense of the iterations tried above. And yes, we will have to address the banner and MetaTools width in a separate, future discussion.

Looks good, thanks for trying out multiple options, Delilah!

**Review by mcbouslog — APPROVED** [review]
LGTM!

Tested on related projects linked FEM Test Project and Daily Minor Planet, also confirmed unrelated projects continue to work as expected (PH TESS, Davy Notebooks).

I appreciate the misc. cleanup as well 👍

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/SVGPanZoom/SVGPanZoom.js [hunk 1]
```diff
@@ -14,14 +15,12 @@ function SVGPanZoom({
   src,
   zooming = true
 }) {
-  const scrollContainer = useRef()
```
[the remark this answers is not in the document]
Comment (goplayoutside3) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/SVGPanZoom/SVGPanZoom.js:17]: Correct, I couldn't find anywhere in SVGPanZoom or its child components where the ref was used.

**Pull request merged**